### PR TITLE
Sign release artifacts

### LINKED ISSUE #180: Sign release artifacts
Extend the [release workflow](#173) by signing the release artifacts.

References #13

### PR DESCRIPTION
**Pull request by RLRabinowitz**
Fixes #180 

Use `cosign` for OIDC-based signing of the release artifacts

- Sign the artifact checksums
- Sign the docker iamges

## Target Release

1.6.0

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -36,10 +44,18 @@ jobs:
...
+
       - name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v4
         with:
           version: v1.20.0
           args: release --clean --timeout=60m --snapshot=${{ !startsWith(github.ref, 'refs/tags/v') }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GPG_FINGERPRINT: ${{ steps.gpg.outputs.fingerprint }}
+          COSIGN_KEY: ${{ secrets.COSIGN_KEY }}
```
Comment (kapilt) [inline .github/workflows/release.yml:60]: wouldn't this be better with tokenless / oidc based signing re not having to worry about a static key/password?

**Comment by RLRabinowitz** [comment]
I will create a follow-up issue to make sure we add the verification process to our documentation website

**Comment by cube2222** [comment]
I'd like for @WSpacelifT to investigate first and confirm whether we want to go with this approach, before we merge this.

File: .goreleaser.yaml [hunk 2]
```diff
@@ -163,6 +163,16 @@ docker_manifests:
 checksum:
   name_template: "{{ .ProjectName }}_{{ .Version }}_SHA256SUMS"
 
+signs:
+  - artifacts: checksum
+    cmd: cosign
+    certificate: "${artifact}.pem"
+    args: ["sign-blob", "--output-certificate=${certificate}", "--output-signature=${signature}", "${artifact}", "--yes"]
```
Comment (WSpacelifT) [inline .goreleaser.yaml:170]: For the first iteration, please ensure that we are not using default settings:
--oidc-issuer string  OIDC provider to be used to issue ID token (default "https://oauth2.sigstore.dev/auth")
but instead, we are acting as a GitHub action as workload identity
--oidc-issuer=https://token.actions.githubusercontent.com

When the value is not defined probably, we are using the default one.

**Review by cube2222 — APPROVED** [review]

File: .goreleaser.yaml (hunk 2 above)
Comment (RLRabinowitz) [reply .goreleaser.yaml:170]: Running the action via GHA makes cosign use the OIDC ID token of the GHA, and https://token.actions.githubusercontent.com/ as the issuer

But I'll make it explicit

**Review by cube2222 — APPROVED** [review]

**Comment by developer-guy** [comment]
Hi guys, I'm happy to see that the RFC @imjasonh started is being implemented so fast! From now on, as OpenTF artifacts are signed, it would be good to have a guide showing how end users can verify them, WDYT?

**Comment by developer-guy** [comment]
I just added OpenTF to OSSF Landscape: https://github.com/ossf/ossf-landscape/pull/333

**Pull request merged**